### ISSUE TITLE
Handle OAuth expired error

### ISSUE THREAD

**Issue by mathias** [heroku/sudo-sandwich#25]
OAuth tokens can expire, which means we'll get back `#<Excon::Response:0x00000000041136a0 @data={:body=>"{\"id\":\"unauthorized\",\"message\":\"Invalid credentials provided.\"}"` when they fail -- (I'd prefer this was an Excon error that got thrown, by using `expects: [200..299]` in the Excon client.)

The old "ecosystem-addon-demo" codebase has a refresher mediator for expired OAuth tokens that we can refer to. (Private repo, so I'm not linking that code here.)

**Comment by jessieay** [comment]
good point, we should not provision the addon if there is an error with the oauth token.
